### PR TITLE
Issue 768

### PR DESCRIPTION
**Pull request by sczembor**
Closes https://github.com/scale-it/algo-builder/issues/768

## Proposed Changes

- Added test scenario for 'uint64' overflow
- Added test for asa creation with total = 2^64-1

### CODE REVIEW

File: packages/runtime/test/src/interpreter/opcode-list.ts [hunk 1]
```diff
@@ -3642,6 +3642,25 @@ describe("Teal Opcodes", function () {
 			assert.equal(5n, stack.pop());
 		});
 
+		it("Int: Should not throw an overflow error", () => {
```
Comment (robert-zaremba) [inline packages/runtime/test/src/interpreter/opcode-list.ts:3645]:
```suggestion
		it("Int: Should work with 2^64 - 1 (max supported number)", () => {
```

File: packages/runtime/test/src/interpreter/opcode-list.ts [hunk 2]
```diff
@@ -3642,6 +3642,25 @@ describe("Teal Opcodes", function () {
...
+			const total = BigInt(2 ** 64) - 1n;
+			assert.doesNotThrow(() => new Int([total.toString()], 0));
+		});
+		it("Int: Should throw an overflow error", () => {
```
Comment (robert-zaremba) [inline packages/runtime/test/src/interpreter/opcode-list.ts:3649]:
```suggestion
		it("Int: Should throw an overflow error on 2^64", () => {
```

File: packages/runtime/test/src/interpreter/opcode-list.ts [hunk 3]
```diff
@@ -3642,6 +3642,25 @@ describe("Teal Opcodes", function () {
...
+			const total = BigInt(2 ** 64);
+			expectRuntimeError(
+				() => new Int([total.toString()], 0),
+				RUNTIME_ERRORS.TEAL.UINT64_OVERFLOW
+			);
+		});
+		it("Int: Should throw an overflow error", () => {
```
Comment (robert-zaremba) [inline packages/runtime/test/src/interpreter/opcode-list.ts:3656]:
```suggestion
		it("Int: Should throw an overflow error on 2^64+1", () => {
```

**Review by robert-zaremba — COMMENTED** [review]
We need to add a test of constructing ASA with total=2^64-1

File: packages/runtime/test/src/interpreter/opcode-list.ts [hunk 4]
```diff
@@ -3642,6 +3642,25 @@ describe("Teal Opcodes", function () {
 			assert.equal(5n, stack.pop());
 		});
 
+		it("Int: Should work with 2^64 - 1 (max supported number)", () => {
+			const total = 2n**64n - 1n;
+			assert.doesNotThrow(() => new Int([total.toString()], 0));
+		});
+		it("Int: Should throw an overflow error on 2^64", () => {
+			const total = BigInt(2 ** 64);
```
Comment (vuvoth) [inline packages/runtime/test/src/interpreter/opcode-list.ts:3650]: Should be `2n**64n` I think

File: packages/runtime/test/src/interpreter/opcode-list.ts [hunk 5]
```diff
@@ -3642,6 +3642,25 @@ describe("Teal Opcodes", function () {
...
+			expectRuntimeError(
+				() => new Int([total.toString()], 0),
+				RUNTIME_ERRORS.TEAL.UINT64_OVERFLOW
+			);
+		});
+		it("Int: Should throw an overflow error on 2^64+1", () => {
+			const total = BigInt(2 ** 64) + 1n;
```
Comment (vuvoth) [inline packages/runtime/test/src/interpreter/opcode-list.ts:3657]: Same here

Comment (sczembor) [reply packages/runtime/test/src/interpreter/opcode-list.ts]: updated it already

File: packages/runtime/test/src/interpreter/opcode-list.ts (hunk 4 above)
Comment (sczembor) [reply packages/runtime/test/src/interpreter/opcode-list.ts]: updated it already

**Review by robert-zaremba — APPROVED** [review]
utACK

**Pull request merged**